fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java

```diff
@@ -190,122 +192,170 @@ protected static Tuple convertOrNull(@Nullable Key.Evaluated key) {
-                        } else {
-                            IndexBuildProto.IndexBuildIndexingStamp savedStamp;
-                            try {
-                                savedStamp = IndexBuildProto.IndexBuildIndexingStamp.parseFrom(bytes);
-                            } catch (InvalidProtocolBufferException ex) {
-                                RecordCoreException protoEx = new RecordCoreException("invalid indexing type stamp",
-                                        LogMessageKeys.INDEX_NAME, common.getIndex().getName(),
-                                        LogMessageKeys.INDEX_VERSION, common.getIndex().getLastModifiedVersion(),
-                                        LogMessageKeys.INDEXER_ID, common.getUuid(),
-                                        LogMessageKeys.EXPECTED, indexingTypeStamp,
-                                        LogMessageKeys.ACTUAL, bytes);
-                                protoEx.initCause(ex);
-                                throw protoEx;
-                            }
-                            if ( savedStamp == null || ! matchingIndexingTypeStamp(savedStamp)) {
-                                throw new RecordCoreException("This index was partly built by another method",
-                                        LogMessageKeys.INDEX_NAME, common.getIndex().getName(),
-                                        LogMessageKeys.INDEX_VERSION, common.getIndex().getLastModifiedVersion(),
-                                        LogMessageKeys.INDEXER_ID, common.getUuid(),
-                                        LogMessageKeys.EXPECTED, indexingTypeStamp,
-                                        LogMessageKeys.ACTUAL, savedStamp);
-                            }
+    private CompletableFuture<Void> setIndexingTypeOrThrow(FDBRecordStore store, boolean continuity) {
+        // continuity is set if this session isn't a continuation of a previous indexing
+        Transaction transaction = store.getContext().ensureActive();
+        IndexBuildProto.IndexBuildIndexingStamp indexingTypeStamp = getIndexingTypeStamp(store);
+        byte[] stampKey = indexBuildTypeSubspace(store, common.getIndex()).getKey();
+        return transaction.get(stampKey)
+                .thenCompose( bytes -> {
+                    if (forceStampOverwrite) {
+                        // no questions asked
+                        transaction.set(stampKey, indexingTypeStamp.toByteArray());
+                        return AsyncUtil.DONE;
+                    }
+                    if (bytes == null) {
+                        if (continuity && indexingTypeStamp.getMethod() !=
+                                          IndexBuildProto.IndexBuildIndexingStamp.Method.BY_RECORDS) {
+                            // backward compatibility - maybe continuing an old BY_RECORD session
+                            return isWriteOnlyButNoRecordScannedOrThrow(store, transaction, stampKey, indexingTypeStamp);
```

Comment:
Is there a test that exercises this code path? I don't see one, but I might have missed it.

It's possible that the only way to test this is to have some thing in the test set up clear out an index type stamp by making direct calls to FDB

Reply:
Added `testIndexFromIndexPersistentContinuePreviousByRecordsWithoutTypeStamp`